install_configure_sshd.sh

```diff
@@ -3,15 +3,20 @@
 if grep -q '#AuthorizedKeysCommand none' "$SSHD_CONFIG_FILE"; then
   sed -i "s:#AuthorizedKeysCommand none:AuthorizedKeysCommand ${AUTHORIZED_KEYS_COMMAND_FILE}:g" "$SSHD_CONFIG_FILE"
 else
-  if ! grep -q "AuthorizedKeysCommand ${AUTHORIZED_KEYS_COMMAND_FILE}" "$SSHD_CONFIG_FILE"; then
+  if ! grep -q "^AuthorizedKeysCommand ${AUTHORIZED_KEYS_COMMAND_FILE}" "$SSHD_CONFIG_FILE"; then
     echo "AuthorizedKeysCommand ${AUTHORIZED_KEYS_COMMAND_FILE}" >> "$SSHD_CONFIG_FILE"
   fi
 fi
 
-if grep -q '#AuthorizedKeysCommandUser nobody' "$SSHD_CONFIG_FILE"; then
-  sed -i "s:#AuthorizedKeysCommandUser nobody:AuthorizedKeysCommandUser nobody:g" "$SSHD_CONFIG_FILE"
-else
-  if ! grep -q 'AuthorizedKeysCommandUser nobody' "$SSHD_CONFIG_FILE"; then
-    echo "AuthorizedKeysCommandUser nobody" >> "$SSHD_CONFIG_FILE"
+if grep -aq 'AuthorizedKeysCommandUser' "$(which sshd)"; then
```

Comment:
If `AuthorizedKeysCommandUser` isn't supported, then we shouldn't set it.

Reply:
shouldn't the file be `$SSHD_CONFIG_FILE` ?

Reply:
My thought here was that we'd check whether the string `AuthorizedKeysCommandUser` exists in the `sshd` binary to determine whether sshd supports the value. I worried checking for it in the config file could cause inconsistency if the config file is managed by something like puppet or ansible and the commented out defaults are stripped out.

I used `grep -a` because I'm searching a binary file.

What are your thoughts about this idea?

Reply:
I see. I didn't payed attention to the `a` flag.